lisa/tools/lsmod.py

```diff
@@ -28,6 +30,22 @@ def _check_exists(self) -> bool:
     def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self._command = "lsmod"
 
+    def get_unloaded_modules(self, modules: Union[str, List[str]]) -> List[str]:
+        if isinstance(modules, str):
+            modules_list = modules.split(" ")
+        else:
+            modules_list = modules
+        assert_that(modules_list).described_as(
+            "Empty modules list passed to get_unloaded_modules in Lsmod tool"
+        ).is_not_empty()
+
+        result = self.run(sudo=True, force_run=True, expected_exit_code=0)
+        minimized_list = []
+        for module in modules_list:
+            if module not in result.stdout:
```

Comment:
if it's possible, check the full string, instead of `in`. It makes some substring is recognized. For example, if "mlx5" in the string, the "mlx" is True.

Reply:
updated, will just leverage run and module_exists to check the list since the result will be cached

Reply:
It fixes, but much slower, if it needs to check multiple modules. Can we still get information from the one time output?

Reply:
Yep I can use a similar pattern to module_exists, commit incoming...